src/GregClient/GregClient.csproj

```diff
@@ -54,8 +55,36 @@
       <Pack>True</Pack>
       <PackagePath>\</PackagePath>
     </None>
+    <Compile Include="..\AssemblyInfoGenerator\AssemblyInfo.cs" Link="Properties\GlobalVersionInfo.cs" />
   </ItemGroup>
-  <ItemGroup Label="Compile time reference assemblies">
-    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.2" PrivateAssets="All" />
+  <ItemGroup>
+    <ProjectReference Include="..\AssemblyInfoGenerator\AssemblyInfoGenerator.csproj">
+      <Private>false</Private>
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>none</IncludeAssets>
+    </ProjectReference>
   </ItemGroup>
+  
+  <Target Name="ReadPackageVersionFromOutputAssemblySingleTfm" Returns="@(PackAssembly)" Condition="'$(IsCrossTargetingBuild)' != 'true'">
```

Comment:
these targets help the automatic package generator to read the version from the AssmeblyInfo.cs file 
https://github.com/dotnet/sdk/issues/9011#issuecomment-357543029